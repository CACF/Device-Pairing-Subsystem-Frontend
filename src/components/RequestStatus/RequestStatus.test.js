import React from 'react';
import {shallow} from 'enzyme';
import RequestStatus from './RequestStatus';

const statusDetails = {
  id: '123',
  icon: 'fa fa-check',
  action: 'Registered',
  showButton: false,
  link: null
}

const  location = {
  pathname: '/request-status',
  state: {details: statusDetails}
}

const historyMock = { push: jest.fn() };

describe('RequestStatus component', () => {

  /* Test if renders correctly */
  test('RequestStatus renders successfully', () => {
    const wrapper = shallow(<RequestStatus location={location} />);
    expect(wrapper.exists()).toBe(true);
  });

  /* Test RequestStatus renders again */
  test('if renders again correctly', () => {
      const wrapper = shallow(<RequestStatus location={location} history={historyMock} />);
      expect(wrapper).toMatchSnapshot();
  });

  /* Test Check class, count html element, text inside DOM */
  test('Check class, count html element, text inside DOM', () => {
      const wrapper = shallow(<RequestStatus location={location} history={historyMock} />);
      //console.log(wrapper.debug());
      expect(wrapper.find('.submitted').length).toEqual(1);
      expect(wrapper.contains(<i className='fa fa-check'></i>)).toBe(true);
      expect(wrapper.find('h4').find('span').text()).toBe('Registered');
      expect(wrapper.find('.msg').find('span').text()).toBe('123');
  });

  /* Test Show button link view request */
  test('View request button link render', () => {
      const  location = {
        pathname: '/request-status',
        state: {
          details: {
            id: '123',
            icon: 'fa fa-check',
            action: 'Registered',
            showButton: true,
            link: '/'
          }
        }
      }
      const wrapper = shallow(<RequestStatus location={location} history={historyMock} />);
      expect(wrapper.find('.link-box').length).toEqual(1);
      expect(wrapper.find('Link')).toHaveLength(1);
  });

})