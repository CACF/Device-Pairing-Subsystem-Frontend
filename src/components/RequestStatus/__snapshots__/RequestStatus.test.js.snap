// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestStatus component if renders again correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RequestStatus
    history={
      Object {
        "push": [MockFunction],
      }
    }
    location={
      Object {
        "pathname": "/request-status",
        "state": Object {
          "details": Object {
            "action": "Registered",
            "icon": "fa fa-check",
            "id": "123",
            "link": null,
            "showButton": false,
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="submitted"
      >
        <div
          className="icon-box"
        >
          <i
            className="fa fa-check"
          />
        </div>
        <h4>
           
          <span>
            Registered
          </span>
           
          !
        </h4>
        <div
          className="msg"
        >
          <p>
             
            <span>
              123
            </span>
          </p>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="icon-box"
          >
            <i
              className="fa fa-check"
            />
          </div>,
          <h4>
             
            <span>
              Registered
            </span>
             
            !
          </h4>,
          <div
            className="msg"
          >
            <p>
               
              <span>
                123
              </span>
            </p>
          </div>,
          false,
        ],
        "className": "submitted",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fa fa-check"
            />,
            "className": "icon-box",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fa fa-check",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              " ",
              <span>
                Registered
              </span>,
              " ",
              undefined,
              "!",
            ],
          },
          "ref": null,
          "rendered": Array [
            undefined,
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Registered",
              },
              "ref": null,
              "rendered": "Registered",
              "type": "span",
            },
            " ",
            undefined,
            "!",
          ],
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
               
              <span>
                123
              </span>
            </p>,
            "className": "msg",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                undefined,
                " ",
                <span>
                  123
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              undefined,
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "123",
                },
                "ref": null,
                "rendered": "123",
                "type": "span",
              },
            ],
            "type": "p",
          },
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="submitted"
        >
          <div
            className="icon-box"
          >
            <i
              className="fa fa-check"
            />
          </div>
          <h4>
             
            <span>
              Registered
            </span>
             
            !
          </h4>
          <div
            className="msg"
          >
            <p>
               
              <span>
                123
              </span>
            </p>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="icon-box"
            >
              <i
                className="fa fa-check"
              />
            </div>,
            <h4>
               
              <span>
                Registered
              </span>
               
              !
            </h4>,
            <div
              className="msg"
            >
              <p>
                 
                <span>
                  123
                </span>
              </p>
            </div>,
            false,
          ],
          "className": "submitted",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="fa fa-check"
              />,
              "className": "icon-box",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fa fa-check",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                undefined,
                " ",
                <span>
                  Registered
                </span>,
                " ",
                undefined,
                "!",
              ],
            },
            "ref": null,
            "rendered": Array [
              undefined,
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Registered",
                },
                "ref": null,
                "rendered": "Registered",
                "type": "span",
              },
              " ",
              undefined,
              "!",
            ],
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p>
                 
                <span>
                  123
                </span>
              </p>,
              "className": "msg",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  undefined,
                  " ",
                  <span>
                    123
                  </span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                undefined,
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "123",
                  },
                  "ref": null,
                  "rendered": "123",
                  "type": "span",
                },
              ],
              "type": "p",
            },
            "type": "div",
          },
          false,
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
